ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/PaywallAction.kt

```diff
@@ -0,0 +1,9 @@
+package com.revenuecat.purchases.ui.revenuecatui.components
+
+import com.revenuecat.purchases.paywalls.components.ButtonComponent
+
+internal sealed interface PaywallAction {
+    object RestorePurchases : PaywallAction
+    object NavigateBack : PaywallAction
+    data class NavigateTo(val destination: ButtonComponent.Destination) : PaywallAction
```

Comment:
Note I did end up reusing `ButtonComponent.Destination` since I felt that was fine.

Reply:
Works for me! 

We could hide the parameter from Java:
```suggestion
    data class NavigateTo(@get:JvmSynthetic val destination: ButtonComponent.Destination) : PaywallAction
```

  
Also, we could think about making it a regular class and using `@Poko` to remove `copy()` and `componentN()`, although I'm not sure if that's necessary as Java users would already get a warning about using `internal` types.